specification/eventgrid/data-plane/Microsoft.Storage/stable/2018-01-01/Storage.json

```diff
@@ -424,6 +424,48 @@
           "type": "string"
         }
       }
+    },
+    "StorageStorageTaskQueuedEventData": {
```

Comment:
We shouldn't repeat Storage twice in the name. Can this simply be StorageTaskQueuedEventData and the event type updated to Microsoft.Storage.TaskQueued?

Reply:
Thank you for the feedback. The intent to use StorageTaskQueued and StorageTaskCompleted is to help tie the event clearly with the actual service (Storage - StorageTasks). Changing to TaskQueued could potentially cause confusion to our customers and they might not immediately recognize the event is related to StorageTasks.

Could you please let me know if you have additional concerns on the event name? We can loop in PM and broader team for the discussion, thanks!

Reply:
Sure, our service name is `StorageTasks`, in Azure Storage there are also other names of `Task` (other services) hence we want to make sure customers are fully aware that this event is associated with the actual feature `StorageTasks`

Reply:
Do you have an example of non-storage tasks that are part of the Azure Storage service?

Reply:
![StorageTasks](https://user-images.githubusercontent.com/128629270/228599601-4919083e-2a32-4db9-880c-6bd8db6073d9.png)
![Tasks](https://user-images.githubusercontent.com/128629270/228599606-658d6117-6c56-465e-b107-be5c5496c348.png)

I attached two pics to display the current naming style shown up in Portal. Under `Data Management` there is our service `Storage Tasks`, which is also the name for those two events. Under `Automation` there is another feature named `Tasks` as well. In the future, there could be other new `Tasks` features and we try to make sure that the event name is unique to the actual feature name.

Reply:
So it seems that when generating, Autorest M4 will automatically correct the event name to have one `Storage`. We could potentially work around this, but it would require customization in each language, so I'm thinking we want to avoid doing this.

Reply:
Updated the PR to use only one `Storage`